### DISCUSSION TITLE
How to use multiple middlewares in Next.js using the middleware.ts file?

### DISCUSSION THREAD

**Discussion by Abdel-Monaam-Aouini** [vercel/next.js#63234] [Help]
### Summary

[How to use multiple middlewares in Next.js using the middleware.ts file?](https://stackoverflow.com/questions/76603369/how-to-use-multiple-middlewares-in-next-js-using-the-middleware-ts-file)

**Comment by mutasim77** [comment]
check this answer: [link](https://github.com/vercel/next.js/discussions/53997#discussioncomment-7407930)

**Reply by rickvdbroek** [reply]
@Abdel-Monaam-Aouini Have a look at my Stackoverflow answer. It is a but bit more advanced including unit tests for the logic but we use it in our enterprise application with more than 1M+ user per day, been running for a few months without problems.

https://stackoverflow.com/questions/77037745/chain-multiple-middleware-functions-in-nextjs-13-for-authentication-and-internat/77886499#77886499

**Comment by z4nr34l** [comment]
Hello, I've committed small package to solve that issues:

https://github.com/z4nr34l/nemo

Version 1.3.2 will come maybe even today:
- adding URL params available in middleware functions just like in pages/layouts but without awaiting them
- better docs and more examples
